akka-remote/src/main/scala/akka/remote/artery/compress/TopHeavyHitters.scala

```diff
@@ -205,6 +219,7 @@ private[remote] final class TopHeavyHitters[T >: Null](val max: Int)(implicit cl
   private def insertKnownNewHeavy(hashCode: HashCodeVal, item: T, count: Long): Unit = {
     removeHash(lowestHitterIndex)
     lowestHitterIndex = insert(hashCode, item, count)
+    takeSnapshot()
```

Comment:
only needs to take an actual snapshot and publish it via the AtomicReference if there is a possibility that the list have changed, which only happens in new insertions. In the future we can subsample this and only take snapshots if a certain number of insertions took place (or, maybe better, a certain total weight worth of updates/insertions is reached compared to the total weight of the table).